piwheels/master/the_scribe.py

```diff
@@ -349,6 +361,76 @@ def write_project_page(self, package):
         except FileExistsError:
             pass
 
+    def delete_package(self, package):
+        """
+        Attempts to remove the index and project page directories (including all
+        known wheel files) of the specified *package*.
+
+        :param str package:
+            The name of the package to delete.
+        """
+        if len(package) == 0:
+            # refuse to delete /simple/ and /project/ by accident
+            self.logger.error('not deleting - package name is empty')
```

Comment:
Nice! Excellent check to make. My one objection is: just an error message in the logs? No, this is a serious condition which should *never* occur and if it does I want to be slapped in the face by it. In other words, throw an exception here and blow up. If this ever happens I want to know about it, to diagnose *how* it came about, and add prevention before we get here.